internal/util/helpers_test.go

```diff
@@ -4,18 +4,767 @@
+		expectedError  string
+		expectedResult util.OpenIdCredentials
+	}{
+		{
+			name:        "Successful load",
+			fileContent: `{"access_token":"access_token","refresh_token":"refresh_token","expiry":"2024-10-05T17:46:27+10:00"}`,
+			expectedResult: util.OpenIdCredentials{
+				AccessToken:          "access_token",
+				RefreshToken:         "refresh_token",
+				AccessTokenExpiresAt: time.Date(2024, 10, 5, 17, 46, 27, 0, time.FixedZone("AEST", 10*60*60)),
+			},
+		},
+		{
+			name:          "Error unmarshaling credentials",
+			fileContent:   `invalid_json`,
+			expectedError: "error unmarshaling credentials: invalid character 'i'",
+		},
+		{
+			name:          "Error reading credentials file",
+			fileContent:   "",
+			expectedError: "error reading credentials file",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // capture range variable
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			// Create a unique temporary directory for the test
+			// tempDir, err := os.MkdirTemp("", "test_load_credentials_"+tt.name)
+			// if err != nil {
+			// 	t.Fatalf("failed to create temp directory: %v", err)
+			// }
+			// defer os.RemoveAll(tempDir)
+
+			tempDir := t.TempDir()
+
+			// Create the minder directory inside the temp directory
+			minderDir := filepath.Join(tempDir, "minder")
+			err := os.MkdirAll(minderDir, 0750)
+			if err != nil {
+				t.Fatalf("failed to create minder directory: %v", err)
+			}
+
+			filePath := filepath.Join(minderDir, "credentials.json")
+
+			if tt.fileContent != "" {
+				// Create a temporary file with the specified content
+				err := os.WriteFile(filePath, []byte(tt.fileContent), 0600)
+				if err != nil {
+					t.Fatalf("failed to write test file: %v", err)
+				}
+				// Print the file path and content for debugging
+				t.Logf("Test %s: written file path %s with content: %s", tt.name, filePath, tt.fileContent)
+			} else {
+				// Print the file path for debugging
+				t.Logf("Test %s: file path %s not created as file content is empty", tt.name, filePath)
+			}
+
+			// Temporarily override the environment variable for the test
+			originalEnv := os.Getenv("XDG_CONFIG_HOME")
+			err = os.Setenv("XDG_CONFIG_HOME", tempDir)
+			if err != nil {
+				t.Errorf("error setting XDG_CONFIG_HOME: %v", err)
+			}
+
+			result, err := util.LoadCredentials()
+			if tt.expectedError != "" {
+				if err == nil || !strings.HasPrefix(err.Error(), tt.expectedError) {
+					t.Errorf("expected error matching %v, got %v", tt.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+				if result.AccessToken != tt.expectedResult.AccessToken || result.RefreshToken != tt.expectedResult.RefreshToken || !result.AccessTokenExpiresAt.Equal(tt.expectedResult.AccessTokenExpiresAt) {
+					t.Errorf("expected result %v, got %v", tt.expectedResult, result)
+				}
+			}
+			defer os.Setenv("XDG_CONFIG_HOME", originalEnv)
+			defer os.RemoveAll(tempDir)
```

Comment:
If you're using `defer` (instead of `t.Cleanup`), you should put it next to the resource it creates. By putting these at the end, you'll only perform the cleanup if the test doesn't exit early (e.g. from `t.Error`)

Reply:
I moved the deferred call as suggested.